Frontend/src/app/clusters/[id]/access-management/page.tsx

```diff
@@ -0,0 +1,222 @@
+import { addUserToCluster, getCluster, removeUserFromCluster } from "@/services/cluster-service";
+import { set_Selected_Cluster } from "@/redux/features/clusterData";
+import { getAllUsers } from "@/services/user-service";
+import { set_Users_Cache } from "@/redux/features/cacheSlice";
+
+interface DataType {
+  key: string;
+  userID: string;
+  Name: string;
+  Email: string;
+  Role: string;
+  Status: boolean;
+}
+
+const PaginationProps = {
+  pageSize: 5,
+  defaultPageSize: 5,
+  showSizeChanger: true,
+  showTotal: (total: number, range: [number, number]) => `${range[0]}-${range[1]} of ${total} items`,
+} as PaginationProps;
+
+export default function AccessManagement({ params }: { params: { id: string } }) {
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState<boolean>(true);
+  const { userData: user } = useAppSelector(state => state.authData)
+  const { selectedCluster } = useAppSelector(state => state.clusterData)
+  const { users } = useAppSelector(state => state.cacheData)
+  const [clusterDetails, setClusterDetails] = useState<IClusterDetails | null>(selectedCluster);
+  const [userData, setUserData] = useState<IUserAccessData[]>(users);
+  const [clusterUsers, setClusterUsers] = useState<IUserAccessData[]>([]);
+
+  const columns: TableProps<DataType>['columns'] = [
+    {
+      title: 'Name',
+      dataIndex: 'Name',
+      key: 'name',
+    },
+    {
+      title: 'Email',
+      dataIndex: 'Email',
+      key: 'email',
+    },
+    {
+      title: 'Role',
+      dataIndex: 'Role',
+      key: 'role',
+    },
+    {
+      title: 'Status',
+      dataIndex: 'Status',
+      key: 'status',
+      render: (_, { Status }) => (
+        <>
+          {Status ? (
+            <Tag color={'green'}>
+              {"Active"}
+            </Tag>) : (
+            <Tag color={"volcano"}>
+              {"Inactive"}
+            </Tag>
+          )}
+        </>
+      ),
+    }
+  ];
+
+  const adminsAccessColumns: TableProps<DataType>['columns'] = [...columns,
+    {
+      title: 'Action',
+      key: 'action',
+      render: (_, record) => (
+        <Space size="middle">
+          <Button type="primary" danger size="small" onClick={() => handleUserRemove(record.userID)}>Remove</Button> 
+        </Space>
+      ),
+    }
+  ];
+
+  const getTableData = () => {
+    return clusterUsers.map((data) => {
+      return {
+        key: data._id,
+        userID: data._id,
+        Name: data.fullName,
+        Email: data.email,
+        Role: data.role,
+        Status: data.enabled,
+      };
+    });
+  }
+
+  const handleUserAdd = async (userID: string) => {
+    const user = userData.find((user) => user._id == userID)
+    setClusterUsers([...clusterUsers, user!])
+    try{
+      const res = await addUserToCluster(userID, clusterDetails!._id);
+      if(res.data){
+        console.log("User added successfully")
+      }
+    }catch(err){
+      console.log("failed to user add")
```

Comment:
Please append the user name to this message

Reply:
user id was added to the log